spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWrite.java

```diff
@@ -386,10 +384,14 @@ protected InternalRowWrapper initPartitionRowWrapper(Types.StructType partitionT
     protected Map<Integer, StructProjection> buildPartitionProjections(
         Types.StructType partitionType, Map<Integer, PartitionSpec> specs) {
       Map<Integer, StructProjection> partitionProjections = Maps.newHashMap();
-      specs.forEach(
-          (specID, spec) ->
-              partitionProjections.put(
-                  specID, StructProjection.create(partitionType, spec.partitionType())));
+
+      for (Map.Entry<Integer, PartitionSpec> entry : specs.entrySet()) {
```

Comment:
Nit: could we save a line by streaming over keySet()?  

for (int specId : specs.keyset()) {
   Partition spec = specs.get(specId);
   ...
}

Itd be nice to do a values transform, but guess there's no clean way here

Reply:
I like it, even though it does an extra lookup. Changed.